### ISSUE TITLE
Don't use `@definition.*` caputere groups

### ISSUE THREAD

**Issue by stsewd** [nvim-treesitter/nvim-treesitter#482]
Currently, we have `@definition.{type}` queries, this doesn't make it easy/obvious to get a list of all definitions (or navigate them). We could use the `#set!` directive instead to set the type of the definition.

Instead of

```
((function_definition
  name: (identifier) @definition.function) @scope
 (#set! definition.function.scope "parent"))
```
We will write it like:

```
((function_definition
  name: (identifier) @definition) @scope
 (#set! definition.scope "parent")
 (#set! definition.type "function"))
```

**Comment by vigoux** [comment]
Not sure, I wanted to use `@definition.type` for actual type inference purpose, to enhance things like completion and so on.

**Comment by steelsojka** [comment]
There should be a function for this in utils that gets all nodes and their type recursively. Take a look at this https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/locals.lua#L108

**Comment by stsewd** [comment]
> There should be a function for this in utils that gets all nodes and their type recursively. Take a look at this https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/locals.lua#L108

yeah, I saw that, but it feels weird that definitions are the only ones using that.
